lib/remote/apilistener.cpp

```diff
@@ -1050,82 +1061,88 @@ bool ApiListener::RelayMessageOne(const Zone::Ptr& targetZone, const MessageOrig
 
 	std::vector<Endpoint::Ptr> skippedEndpoints;
 
-	bool relayed = false, log_needed = false, log_done = false;
-
-	std::set<Endpoint::Ptr> targetEndpoints;
-
+	std::set<Zone::Ptr> allTargetZones;
 	if (targetZone->GetGlobal()) {
-		targetEndpoints = localZone->GetEndpoints();
-
+		/* if the zone is global, the message has to be relayed to our local zone and direct children */
+		allTargetZones.insert(localZone);
 		for (const Zone::Ptr& zone : ConfigType::GetObjectsByType<Zone>()) {
-			/* Fetch immediate child zone members */
 			if (zone->GetParent() == localZone) {
-				std::set<Endpoint::Ptr> endpoints = zone->GetEndpoints();
-				targetEndpoints.insert(endpoints.begin(), endpoints.end());
+				allTargetZones.insert(zone);
 			}
 		}
 	} else {
-		targetEndpoints = targetZone->GetEndpoints();
+		/* whereas if it's not global, the message is just relayed to the zone itself */
+		allTargetZones.insert(targetZone);
 	}
 
-	for (const Endpoint::Ptr& targetEndpoint : targetEndpoints) {
-		/* Don't relay messages to ourselves. */
-		if (targetEndpoint == localEndpoint)
-			continue;
+	bool needsReplay = false;
 
-		log_needed = true;
+	for (const Zone::Ptr& currentTargetZone : allTargetZones) {
+		bool relayed = false, log_needed = false, log_done = false;
 
-		/* Don't relay messages to disconnected endpoints. */
-		if (!targetEndpoint->GetConnected()) {
-			if (targetZone == localZone)
-				log_done = false;
+		for (const Endpoint::Ptr &targetEndpoint : currentTargetZone->GetEndpoints()) {
```

Comment:
2. Feel free to use `auto&`.

Reply:
I'd stick to the type names. They weren't too much work to type out and I think it's clearer this way.